test/e2e_node/cpu_manager_test.go

```diff
@@ -0,0 +1,424 @@
+	}
+}
+
+func getLocalNodeCPUDetails(f *framework.Framework) (cpuCapVal int64, cpuAllocVal int64, cpuResVal int64) {
+	localNodeCap := getLocalNode(f).Status.Capacity
+	cpuCap := localNodeCap[v1.ResourceCPU]
+	localNodeAlloc := getLocalNode(f).Status.Allocatable
+	cpuAlloc := localNodeAlloc[v1.ResourceCPU]
+	cpuRes := cpuCap.Copy()
+	cpuRes.Sub(cpuAlloc)
+
+	// RoundUp reserved CPUs to get only integer cores.
+	cpuRes.RoundUp(0)
+
+	return cpuCap.Value(), (cpuCap.Value() - cpuRes.Value()), cpuRes.Value()
+}
+
+func waitForContainerRemoval(ctnPartName string) {
+	Eventually(func() bool {
+		err := exec.Command("/bin/sh", "-c", fmt.Sprintf("if [ -n \"$(docker ps -a | grep -i %s)\" ]; then exit 1; fi", ctnPartName)).Run()
+		if err != nil {
+			return false
+		}
+		return true
+	}, 2*time.Minute, 1*time.Second).Should(BeTrue())
+}
+
+func isHTEnabled() bool {
+	err := exec.Command("/bin/sh", "-c", "if [[ $(lscpu | grep \"Thread(s) per core:\" | cut -c24) != \"2\" ]]; then exit 1; fi").Run()
+	if err != nil {
+		return false
+	}
+	return true
+}
+
+func getCPUSiblingList(cpuRes int64) string {
+	out, err := exec.Command("/bin/sh", "-c", fmt.Sprintf("cat /sys/devices/system/cpu/cpu%d/topology/thread_siblings_list | tr -d \"\n\r\"", cpuRes)).Output()
+	framework.ExpectNoError(err)
+	return string(out)
+}
+
+func setOldKubeletConfig(f *framework.Framework) {
+	if oldCfg != nil {
+		framework.ExpectNoError(setKubeletConfiguration(f, oldCfg))
+	}
+}
+
+func setupForTests(f *framework.Framework) {
+	// Run only if the container runtime is Docker.
+	framework.RunIfContainerRuntimeIs("docker")
+
+	// Enable CPU Manager in Kubelet with static policy.
+	oldCfg, err := getCurrentKubeletConfig()
+	framework.ExpectNoError(err)
+	clone, err := scheme.Scheme.DeepCopy(oldCfg)
+	framework.ExpectNoError(err)
+	newCfg := clone.(*kubeletconfig.KubeletConfiguration)
+
+	// Enable CPU Manager using feature gate.
+	if newCfg.FeatureGates != "" {
+		newCfg.FeatureGates = fmt.Sprintf("%s,%s", cpuManagerFeatureGate, newCfg.FeatureGates)
+	} else {
+		newCfg.FeatureGates = cpuManagerFeatureGate
+	}
+
+	// Set the CPU Manager policy to static.
+	newCfg.CPUManagerPolicy = cpuManagerPolicy
+
+	// Set the CPU Manager reconcile period to 1 second.
+	newCfg.CPUManagerReconcilePeriod = metav1.Duration{Duration: 1 * time.Second}
+
+	// The Kubelet panics if either kube-reserved or system-reserved is not set
+	// when CPU Manager is enabled. Set cpu in kube-reserved > 0 so that
+	// kubelet doesn't panic.
+	if _, ok := newCfg.KubeReserved["cpu"]; !ok {
+		newCfg.KubeReserved["cpu"] = "200m"
+	}
+	// Update the Kubelet configuration.
+	framework.ExpectNoError(setKubeletConfiguration(f, newCfg))
+
+	// Wait for the Kubelet to be ready.
+	Eventually(func() bool {
+		nodeList := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+		return len(nodeList.Items) == 1
+	}, time.Minute, time.Second).Should(BeTrue())
+
+	cpuCap, cpuAlloc, cpuRes = getLocalNodeCPUDetails(f)
```

Comment:
Can we avoid making these package globals by moving this call into `runCPUManagerTests()`?